### PR TITLE
ci: update migration tests with updated versions, version checks and env paths

### PR DESCRIPTION
**Pull request by SwaySway**
#### Description of changes
Removing v4 migration tests as lambda functions with node12 runtimes are now deprecated.
Removing latest tests and replaced them with `v6.1.0` migration tests.
Updated `split-e2e-tests` to reflect updated migration tests and to pass the migrate from version into `AMPLIFY_PATH`
Updated migration tests to also include workflow changes from the following prs
 - https://github.com/aws-amplify/amplify-cli/pull/8287
 - https://github.com/aws-amplify/amplify-cli/pull/8806

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by SwaySway** [comment]
cloud-e2e run: https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/4742/workflows/ea3c781b-0e46-45fc-be66-6881d4d07ec8

File: packages/amplify-migration-tests/src/migration-helpers/check-version.ts [hunk 1]
```diff
@@ -13,3 +13,12 @@ export function versionCheck(cwd: string, testingWithLatestCodebase = false, ver
       });
   });
 }
+
+/**
+ * This list is used to check migration tests with the following changes. (excludes layer migrationi tests)
```
Comment (danielleadams) [inline packages/amplify-migration-tests/src/migration-helpers/check-version.ts:18]:
```suggestion
 * This list is used to check migration tests with the following changes. (excludes layer migration tests)
```

**Review by edwardfoyle — APPROVED** [review]

File: packages/amplify-migration-tests/src/__tests__/update_tests/function_migration_update.test.ts [hunk 2]
```diff
@@ -116,6 +121,7 @@ describe('amplify function migration', () => {
       runtimes: [runtime],
     };
 
+    // toggle to latest
```
Comment (danielleadams) [inline packages/amplify-migration-tests/src/__tests__/update_tests/function_migration_update.test.ts:124]: Does `// toggle to latest` refer to the `testingWithLatestCodebase` flag? If so - is this comment needed here?

File: packages/amplify-migration-tests/src/__tests__/update_tests/api_migration_update.test.ts [hunk 3]
```diff
@@ -12,35 +10,54 @@ import {
...
     // init the project and add api with installed cli
-    await initJSProjectWithProfile(projRoot, { name: projectName });
-    await addApiWithoutSchema(projRoot);
-    await updateApiSchema(projRoot, projectName, initialSchema);
+    const { projectName } = getProjectConfig(projRoot);
+    await addApiWithoutSchemaOldDx(projRoot);
+    updateApiSchema(projRoot, projectName, initialSchema);
     await amplifyPush(projRoot);
+
+    // toggle to latest version
```
Comment (danielleadams) [inline packages/amplify-migration-tests/src/__tests__/update_tests/api_migration_update.test.ts:60]: same here

**Review by danielleadams — COMMENTED** [review]
Just a few nit comments, but looks good to merge once they are addressed.

File: scripts/split-e2e-tests.ts [hunk 4]
```diff
@@ -471,17 +478,19 @@ function main(): void {
   );
   const splitV4MigrationTests = splitTests(
```
Comment (jhockett) [inline scripts/split-e2e-tests.ts:479]: nit: rename function

```suggestion
  const splitV5MigrationTests = splitTests(
```

**Review by jhockett — APPROVED** [review]

**Pull request merged**